File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/inbox_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'context.dart';

class CrawChatInboxModule {
  final CrawChatSdkContext context;

  CrawChatInboxModule(this.context);

  Future<InboxResponse?> list() {
    return context.backendClient.inbox.getInbox();
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/inbox-module.ts
import type { InboxResponse } from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatInboxModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  list(): Promise<InboxResponse> {
    return this.context.backendClient.inbox.getInbox();
  }
}
